src/common/transformations/src/transformations/sdpa_to_paged_attention/total_sequence_length_pattern.cpp

```diff
@@ -29,11 +30,13 @@ ov::pass::TotalSequenceLengthPattern::TotalSequenceLengthPattern(
         //  use symbolic infra or look at the constant input
         auto gather = m.get_match_root();
         auto target_type = gather->get_output_element_type(0);
-        std::shared_ptr<Node> replacement;
-        if (max_context_len->get_output_element_type(0) != target_type) {
-            replacement = std::make_shared<v0::Convert>(max_context_len, target_type);
-        } else {
-            replacement = max_context_len;
+        std::shared_ptr<Node> replacement = max_context_len;
+        if (replacement->get_output_element_type(0) != target_type) {
+            replacement = std::make_shared<v0::Convert>(replacement, target_type);
+        }
+        auto required_shape = gather->get_output_partial_shape(0);
+        if (replacement->get_output_partial_shape(0) != required_shape && required_shape.rank().is_static()) {
```

Comment:
minor: It's better to change order of the checks:

required_shape.rank().is_static() && replacement->get_output_partial_shape(0) != required_shape